Improve character#show UI, add character descriptions

Throne3d opened the pull request:
This is a rather large change, I'm afraid. No new tests because it's mostly (all?) UI and cucumber is scary and complicated. It might be worth checking out the branch and checking things work as expected.

- Extract the style of the new object pages into CSS; use appropriate classes (icons and users)
- Update the `characters#show` page to use the new interface
- Move the character template to the infobox, use `cycle` instead of manually setting even/odd classes
- Add a link to the character in the breadcrumbs
- Shrink the width (from 200px to 175px) of the character info box
- Add character descriptions to both `characters#edit` and `characters#show`

## Screenshots:
**Icons and users display as before**
![image](https://cloud.githubusercontent.com/assets/577128/23629966/3ead7d5e-02b1-11e7-81e2-abc2b0401850.png)
![image](https://cloud.githubusercontent.com/assets/577128/23629976/47d914a6-02b1-11e7-8298-b1425e9aee0d.png)

**The new characters interface**
![image](https://cloud.githubusercontent.com/assets/577128/23630072/89e0be62-02b1-11e7-95b9-25718f4d57e6.png)
**Galleries**
![image](https://cloud.githubusercontent.com/assets/577128/23630081/9637aa54-02b1-11e7-94df-118f5b8cf8c2.png)
**Posts**
![image](https://cloud.githubusercontent.com/assets/577128/23630092/a0bacac4-02b1-11e7-9508-54f526e6b0f8.png)
**Editor** (it uses a textarea, does not use TinyMCE)
![image](https://cloud.githubusercontent.com/assets/577128/23630174/f068619e-02b1-11e7-90fe-7220d78ba23b.png)
**New character page**
![image](https://cloud.githubusercontent.com/assets/577128/23630285/7aa76d1e-02b2-11e7-9656-3020bbf12b69.png)
**Character with lots of missing details**
![image](https://cloud.githubusercontent.com/assets/577128/23630313/92c99656-02b2-11e7-9941-2326467701d9.png)
**Logged-out views**
![image](https://cloud.githubusercontent.com/assets/577128/23630333/a43fbae6-02b2-11e7-8a90-184940d7aac9.png)
![image](https://cloud.githubusercontent.com/assets/577128/23630362/c1326c70-02b2-11e7-8714-30e9da3664ab.png)

Everything seems to work fine, I think?

Throne3d commented:
Resolves part of #1.

Marri commented:
Too tired to go through this properly right now but a) I suggest the .vtop class to fix the weird centered Description labels and b) should we maybe put more thought into which if any of the fields display when empty and which don't?

That all aside it looks lovely from the screenshots and I appreciate you very much.

Throne3d commented:
<3

I wasn't sure which of Nickname(s), Facecast, Setting to hide – they're reasonably frequently blank but having them *all* be blank (and hiding them all) would be weird, and I'd probably need to display a "No extra info" bar or something? I wasn't sure how to go around it.

Throne3d commented:
![image](https://cloud.githubusercontent.com/assets/577128/23716807/6edf632e-0429-11e7-80d3-d754fd174862.png)
Rebased against master, added `vtop` to the description header, added a style for the textarea on new character (the previous style only applied to edit character).

I'm still not sure which of Nickname(s), Facecast and Setting to hide when blank – if all of them, something should presumably be displayed when they're all blank, but again I'm not sure what. The present configuration might work okay? Or possibly only "Facecast" shows when blank, since that's very rarely going to be blank, whereas the others might be.

Throne3d commented:
I actually went ahead and hid Nickname(s) and Setting when blank, since that seems to make sense.

Again, character with no additional information (just name):
![image](https://cloud.githubusercontent.com/assets/577128/23717068/51d03ffa-042a-11e7-84bb-e0686344b486.png)

Marri commented on app/views/posts/_list.haml:
```diff
@@ -6,7 +6,7 @@
 %span.time-loaded= pretty_time(DateTime.now.in_time_zone)
 
-%table{style: local_assigns[:style].to_s}
+%table{style: local_assigns[:style].to_s, class: local_assigns[:table_class].to_s}
```
You can take out the style one, grep says it's only icons and users using it. (Using `ag 'partial.*list.*style'`)
